product_variant_supplierinfo/models/procurement.py

```diff
@@ -15,14 +15,17 @@ def _get_product_supplier(self, procurement):
         """returns the main supplier of the procurement's product
            given as argument"""
         product = procurement.product_id
+        seller = product.suspend_security().seller_id
         company_supplier = self.env['product.supplierinfo'].search([
             '|',
             '&',
             ('product_tmpl_id', '=', product.product_tmpl_id.id),
             ('product_id', '=', False),
             ('product_id', '=', product.id),
             ('company_id', '=', procurement.company_id.id),
-            ], limit=1)
+        ], limit=1)
         if company_supplier:
             return company_supplier.name
-        return procurement.product_id.seller_id
+        elif (seller.company_id == procurement.company_id or
+                not seller.company_id):
+            return procurement.product_id.seller_id
```

Comment:
OK, I see, thanks for the explanation. A comment in code in the `elif` would be good for future references.